sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuMultiFileReader.scala

```diff
@@ -114,6 +114,24 @@ object MultiFileThreadPoolUtil {
   }
 }
 
+/** A factory class to get a thread pool */
+class MultiFileThreadPoolFactory {
+  private var threadPool: Option[ThreadPoolExecutor] = None
+
+  private def initThreadPool(
+      threadTag: String,
+      numThreads: Int): ThreadPoolExecutor = synchronized {
+    if (threadPool.isEmpty) {
```

Comment:
Nit: Why is this checking for an initialized threadPool?  It's a private method called only in a `getOrElse` context, which means we know it won't be called unless threadPool is empty.  Either that or we should just ditch `initThreadPool` and have its implementation be the code for `getThreadPool`.

Reply:
This piece of code is not written by me (I just lifted it to a parent class to avoid duplicate code in the 3 format readers), however I think this is a Double-Checked Locking which could get a better perf if `getThreadPool` is called quite much.